Yoast/wordpress-seo
# Allow removing an author archive from search results

## Summary

This PR can be summarized in the following changelog entry:

* Removes the option to exclude an author from the XML sitemap in favor of a broader option. Instead we now have an option to set whether this author's archive should be visible in the search results. If you choose not to allow this archive in the search results, it's also excluded from the author sitemap.

## Relevant technical choices:

* The old user meta key `wpseo_excludeauthorsitemap` is replaced in the upgrade function by `wpseo_noindex_author`.

## Test instructions

This PR can be tested by following these steps:

* Go to a user's author archive, view source, no `noindex` should be set.
* Go to the XML author sitemap, the author should appear.
* Go to the user's profile in the admin, set to not allow in search results.
* The user's author archive page should now have a meta robots `noindex`.
* The user should disappear from the XML author sitemap.

## Quality assurance


Fixes #6086
